File: zema_store_backend/src/models/follower.ts
import mongoose, { Schema, Document } from "mongoose";

export interface IFollowerDocument extends Document {
  artistId: mongoose.Schema.Types.ObjectId;
  customerId: mongoose.Schema.Types.ObjectId;
}

const followerSchema: Schema<IFollowerDocument> = new Schema({
  artistId: {
    type: mongoose.Schema.Types.ObjectId,
    required: true,
  },
  customerId: {
    type: mongoose.Schema.Types.ObjectId,
    required: true,
  },
});

const Follower = mongoose.model<IFollowerDocument>("Follower", followerSchema);

export default Follower;

File: zema_store_mobile/lib/models/follower.dart
class Follower {

  Follower({
    required this.artistId,
    required this.customerId
  });
  final String artistId;
  final String customerId;

  factory Follower.fromJson(Map<String,dynamic> json){
    return Follower(
      artistId: json['artistId'], 
      customerId: json['customerId']
    );
  }

}